### PR TITLE
Incubate mechanic initial support

### LINKED ISSUE #2775: MOM: Incubate Mechanic

### PR DESCRIPTION
**Pull request by Northmoc**
Things that probably still need doing (may not fit this PR)
 - IncubateAI for AI decision making... could probably be based on AmassAI a little?
 - `Fetch due to missing key:  for Phyrexian Token (135)` (image issue)

Closes #2775

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/glissa_herald_of_predation.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Name:Glissa, Herald of Predation
+ManaCost:3 B G
+Types:Legendary Creature Phyrexian Zombie Elf
+PT:3/5
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigCharm | TriggerZones$ Battlefield | TriggerDescription$ At the beginning of combat on your turn, ABILITY
+SVar:TrigCharm:DB$ Charm | Choices$ Incubate,Transform,PumpAll | CharmNum$ 1
+SVar:Incubate:DB$ Incubate | Amount$ 2 | Times$ 2 | SpellDescription$ Incubate 2 twice. (To incubate 2, create an Incubator token with three +1/+1 counters on it and "{2}: Transform this artifact." It transforms into a 0/0 Phyrexian artifact creature.)
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/glissa_herald_of_predation.txt:7]: three = two

File: forge-gui/res/cardsfolder/upcoming/norns_inquisitor.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Name:Norn's Inquisitor
+ManaCost:1 W
+Types:Creature Phyrexian Knight
+PT:1/1
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigIncubate | TriggerDescription$ When CARDNAME enters the battlefield, incubate 2.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/norns_inquisitor.txt:5]: Missing 2nd trigger

**Comment by Hanmac** [comment]
depending on the rules of "Incubate N", we probably need to put the "M times" around the function like done with "Investigate"

The difference is "creating M tokens with N counters each" and "create 1 Token with N counters M times"

matters for replacement effects and trigger

File: forge-gui/res/cardsfolder/upcoming/glissa_herald_of_predation.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/glissa_herald_of_predation.txt:7]: Description too :P

File: forge-game/src/main/java/forge/game/ability/effects/IncubateEffect.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        final int times = AbilityUtils.calculateAmount(card, sa.getParamOrDefault("Times", "1"), sa);
+
+        CardZoneTable triggerList = new CardZoneTable();
+        MutableBoolean combatChanged = new MutableBoolean(false);
+
+        sa.putParam("WithCountersType", "P1P1");
+        sa.putParam("WithCountersAmount", sa.getParamOrDefault("Amount", "1"));
+
+        for (int i = 0; i < times; i++) {
+            makeTokenTable(makeTokenTableInternal(activator, "incubator", 1, sa), false,
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/ability/effects/IncubateEffect.java:48]: you should probably move triggerList and combatChanged into the loop too
like done with in `InvestigateEffect`

they probably can't change the combat anyway, but 🤷‍♂️

File: forge-game/src/main/java/forge/game/ability/effects/ChangeZoneEffect.java [hunk 4]
```diff
@@ -603,6 +603,17 @@ private void changeKnownOriginResolve(final SpellAbility sa) {
                             // If it can't Transform, don't change zones.
                             continue;
                         }
+                    } else if (sa.hasParam("FrontFace")){
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/ability/effects/ChangeZoneEffect.java:606]: hmn was that really needed?

if you exile it the engine should already make sure it doesn't stack on its backside

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/ability/effects/ChangeZoneEffect.java]: Yeah, I just didn't test things properly before adding that param

**Review by tool4ever — APPROVED** [review]
nice job

**Pull request merged**